File: 현세미/프로그래머스/문자열압축.py
def solution(s):
    n = len(s)
    min_length = n
    
    for unit in range(1, n // 2 + 1):
        compressed = ""
        prev = s[:unit]
        count = 1
        
        for i in range(unit, n, unit):
            if s[i:i+unit] == prev:
                count += 1
            else:
                compressed += (str(count) + prev) if count > 1 else prev
                prev = s[i:i+unit]
                count = 1
        
        compressed += (str(count) + prev) if count > 1 else prev
        min_length = min(min_length, len(compressed))
    
    return min_length

File: 이진영/프로그래머스/문자열 압축.js
function solution(s) {
    let min_length = s.length;
    
    for (let i = 1; i < s.length; i++) { // 크기 1부터 문자열 길이 - 1까지
        min_length = Math.min(min_length, press(s, i).length);
    }
    
    return min_length;
}

// 압축 함수
const press = (str, n) => {
    const slice = Math.ceil(str.length / n); // 이터레이션 횟수
    let pressed = "" // 반환할 압축한 문자열
    let prev = ""; // 전 문자열 확인
    let count = 0 // 갯수 확인
    
    for (let i = 0; i < slice; i++) {
        const part = str.slice(i * n, n * (i + 1));
        if (prev === part) {
            count++;
            continue;
        }
        
        if (prev) {
            pressed += count === 1 ? prev : count.toString() + prev;
        }
        
        prev = part;
        count = 1;
    }
    
    pressed += count === 1 ? prev : count.toString() + prev; // 마지막 처리
    
    return pressed;
}